### ISSUE TITLE
Compatible when use angular 2 and @types

### ISSUE THREAD

**Issue by eternalsolar** [Microsoft/TypeScript#11032]
Hi,
I have upgrade my app to angular 2 and try to use @types instead of typings.
But it seem I can not import \* as _ from "path", IDE not recognize the library, I can just import specify namespace from "path", for example, BootstrapSwitch from '@types/bootstrap-switch'.

Then, when I run build.prod, the build.prod file alert error that

> error TS2306: File 'E:/Duy/Work/Git/OrchardCredit/Frontend/node_modules/@types/bo
> otstrap-switch/index.d.ts' is not a module.

Any help ?

**Comment by nippur72** [comment]
`bootstrap-switch` doesn't seem to be a module, try importing its typing with:
```ts
<reference types="bootstrap-switch" />
```

**Comment by mhegazy** [comment]
has @nippur72's suggestion solved the issue?

**Comment by eternalsolar** [comment]
Of course If I use typings like his suggestion, add into the typings.json, everything works well.
But I mean i'm changing from using typings to the new way @types.
So I run npm install -save @types/bootstrap-switch. then in the node_modules, the folder @types with subfolders bootstrap-switch has appeared.
Then I came to my angular2 components and try to use 
`import * as _ from 'bootstrap-switch` 
this way show warning can not resolved bootstrap-switch so I change to
`import * as _ from '@types/bootstrap-switch` 
and no warning from IDE.

After that I called some method and still got no warning from IDE, but when I run build.prod, this error

> error TS2306: File 'E:/Duy/Work/Git/OrchardCredit/Frontend/node_modules/@types/bo
otstrap-switch/index.d.ts' is not a module.

appeared.

**Comment by mhegazy** [comment]
> this way show warning can not resolved bootstrap-switch so I change to
> `import * as _ from '@types/bootstrap-switch' `

you should not do this. this is not going to work at runtime.

> After that I called some method and still got no warning from IDE, but when I run build.prod, this error

To explain, the way the [`bootstrap-switch` declaration file](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/types-2.0/bootstrap-switch/index.d.ts) is authored does not expose a module called `"bootstrap-switch"`. From the documentation there does not seem to be possible either.
if you believe this should work, then please file a PR against DefinitlyTyped with the addition of a module declaration to the file:

````ts
declare module "bootstrap-switch" {
    export = BootstrapSwitch;
}
```

What the error is telling you, is that this file is not a module, and only modules can be imported. this file is meant to be used as a global variable.

**Comment by nippur72** [comment]
So to sum up:
- install package with `npm i bootstrap-switch` (as before)
- install types with `npm i @types\bootstrap-switch` 
- since it's not a module load it globally (e.g.`<script>`) from `node_modules\bootstrap-switch`
- in typescript, use types with `<reference types="bootstrap-switch" />` 

What you gain from the new syntax is that `<reference />` does not have to specify the full path, but only just the package name. A little advantage that makes everything looks cleaner. Of course the big gain is when what you are loading is a true module.

**Comment by nomaed** [comment]
I am trying to figure this out as well.
* Installed `angular` and `@types/angular` to node_modules
* Used `import * as angular from "angular"` in the `ts` file: ` error TS2307: Cannot find module 'angular'.`
* Added to `tsconfig.json`: `paths: { "angular": "node_modules/@types/angular/index.d.ts" }`, still getting the same error.

If I remove the import statement, compilation works file and in the list of files I see that `@types/angular` (and `@types/jquery`) is being included automatically, without me asking for them.
   
The next step was `npm install --save @types/angular-resource`.

Now, running `tsc`, I see that this file is being added automatically to the build process, *AND* the compiler is shouting:
```
9 import * as angular from 'angular';
                           ~~~~~~~~~

node_modules/@types/angular-resource/index.d.ts(9,26): error TS2307: Cannot find module 'angular'.


192             get(name: '$resource'): ng.resource.IResourceService;
                                           ~~~~~~~~

node_modules/@types/angular-resource/index.d.ts(192,40): error TS2305: Module 'angular' has no exported member 'resource'.

/usr/local/lib/node_modules/typescript/lib/lib.es6.d.ts
/Users/.../git/ts2test/src/index.ts
/Users/.../git/ts2test/node_modules/@types/jquery/index.d.ts
/Users/.../git/ts2test/node_modules/@types/angular/index.d.ts
/Users/.../git/ts2test/node_modules/@types/angular-resource/index.d.ts
```

Adding or removing `paths` in `tsconfig.json` makes no difference whatsoever.

I must be missing something, but currently I am at a loss.

```
$ tsc -v
Version 2.0.3
```
``` json
$ cat tsconfig.json
{
    "compilerOptions": {
        "module": "es2015",
        "target": "es6",
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "strictNullChecks": true,
        "noImplicitAny": true,
        "listFiles": true,
        "sourceMap": true,
        "declaration": false,
        "pretty": true,
        "outDir": "built"
    },
    "paths": {
        "angular": "node_modules/@types/angular/index.d.ts",
        "jquery": "node_modules/@types/jquery/index.d.ts"
    },
    "lib": [
        "es5",
        "es2015",
        "es2016"
    ],
    "files": [
        "src/index"
    ]
}
```
```typescript
$ cat src/index.ts
// Test importing a module
//import * as angular from "angular";

angular.module('foo', []);
```

All of it is in a repo:
https://github.com/nomaed/ts2test

**Comment by nippur72** [comment]
@nomaed look at issues in your repo, I replied there.

**Comment by nomaed** [comment]
Thank you! It was a quick fix for me being confused ;)
I replied @ https://github.com/nomaed/ts2test/issues/1

**Comment by reppners** [comment]
Got the same problem trying to migrate from typings to @types. No import of any namespace is working.

tsconfig.json
```JSON
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es5",
    "sourceMap": true,
    "pretty": true,
    "stripInternal": true,
    "noEmitHelpers": false,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "removeComments": true,
    "preserveConstEnums": false,
    "noImplicitAny": true,
    "strictNullChecks": true,
    "noImplicitReturns": true,
    "noImplicitThis": true,
    "forceConsistentCasingInFileNames": true,
    "suppressExcessPropertyErrors": false,
    "suppressImplicitAnyIndexErrors": false,
    "noUnusedParameters": true,
    "noUnusedLocals": true,
    "outDir": "tmp-out",
    "baseUrl": "./jspm_packages",
    "paths": {
      "ng-metadata/*": [
        "npm/ng-metadata@2.2.0/*"
      ],
      "rxjs/*": [
        "npm/rxjs@5.0.0-beta.12/*"
      ],
      "drag-drop-polyfill/*": [
        "npm/drag-drop-polyfill@2.0.0-beta.2/release/*"
      ]
    }
  },
  "exclude": [
    "node_modules",
    "jspm_packages",
    "tests",
    "e2e-tests",
    "theming"
  ]
}
```

sample code
```TS
import IRootScopeService = angular.IRootScopeService;
```

error messages
`Error:(9, 36) TS2305: Module '"..../node_modules/@types/angular/index".angular' has no exported member 'IRootScopeService'.`
`Error:(9, 36) TS2339: Property 'IRootScopeService' does not exist on type 'IAngularStatic'.`

But for me the problem is not angular-specific, there are also other typings that are not working in the same way.

```TS
import DSInstanceShorthands = JSData.DSInstanceShorthands;
```
`Error:(8, 31) TS2503: Cannot find namespace 'JSData'.`

 I also tried `"moduleResolution": "node"` but that didn't change anything.

**Comment by akankshagaur** [comment]
Hi, I am having the same problem. Is anyone able to resolve this issue?

**Comment by nomaed** [comment]
@akankshagaur I reverted to using Typings included from tsconfig.json

**Comment by akankshagaur** [comment]
@nomaed finally, my issue was resolved. check [here](http://stackoverflow.com/questions/40124717/angular-2-moved-to-types-giving-error-cannot-find-name-require/40124889#40124889) make sure your global n local typescript versions are same. ideally it should pick up from node_modules/@types folder but it wasn't working at my end. I added types:[] for all my typings in tsconfig.json.. no need for typeRoots
